sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -585,22 +586,17 @@ private static boolean isPrimitiveType(DetailAST ast) {
     /**
      * Checks whether token is array or elipsis.
      *
-     * @param identToken
+     * @param typeToken
      *        the token
      * @return true if yes
      */
-    private static boolean isArrayOrElipsis(final DetailAST identToken) {
-        final DetailAST next = identToken.getNextSibling();
-        final boolean result;
-        switch (next.getType()) {
-            case TokenTypes.ARRAY_DECLARATOR:
-            case TokenTypes.ELLIPSIS:
-                result = true;
-                break;
-            default:
-                result = false;
-        }
-        return result;
+    private static boolean isArrayOrElipsis(final DetailAST typeToken) {
+        final DetailAST next = typeToken.getNextSibling();
+        final boolean isArrayDeclarator =
+                typeToken.findFirstToken(TokenTypes.ARRAY_DECLARATOR) != null;
+        final boolean hasArrayOrEllipses =
+                TokenUtil.isOfType(next, TokenTypes.ARRAY_DECLARATOR, TokenTypes.ELLIPSIS);
+        return hasArrayOrEllipses || isArrayDeclarator;
```

Comment:
Switch was overkill for two branches, refactored this and modified logic a bit for new `type` ast.